Sprint-2/stretch/count-words.js

```diff
@@ -1,28 +1,31 @@
-/*
-  Count the number of times a word appears in a given string.
+function countWords(str) {
+  const cleanedStr = str.replace(/[.,!?]/g, "").toLowerCase();
 
-  Write a function called countWords that
-    - takes a string as an argument
-    - returns an object where
-          - the keys are the words from the string and
-          - the values are the number of times the word appears in the string
+  
+  const words = cleanedStr.split(/\s+/);
 
-  Example
-  If we call countWords like this:
 
-  countWords("you and me and you") then the target output is { you: 2, and: 2, me: 1 }
+  const wordCounts = {};
 
-  To complete this exercise you should understand
-    - Strings and string manipulation
-    - Loops
-    - Comparison inside if statements
-    - Setting values on an object
 
-## Advanced challenges
+  for (const word of words) {
+    if (wordCounts[word]) {
+      wordCounts[word] += 1;
+    } else {
+      wordCounts[word] = 1;
+    }
+  }
 
-1. Remove all of the punctuation (e.g. ".", ",", "!", "?") to tidy up the results
 
-2. Ignore the case of the words to find more unique words. e.g. (A === a, Hello === hello)
+  const sortedWordCounts = Object.entries(wordCounts)
+    .sort(([, countA], [, countB]) => countB - countA)
+    .reduce((sortedObj, [key, value]) => {
+      sortedObj[key] = value;
+      return sortedObj;
+    }, {});
```

Comment:
There is also a built-in method in `Object` that does the opposite of `.entries()`.